### PR TITLE
5.0b3 1.0

### PR DESCRIPTION
**Pull request by FriedrichAltheide**
**Done**
- Show send button
- Improve omemo handling for broken sessions
- Increase startup speed
- Fix trust UI. Show trust stage. ToFU, trusted, untrusted
- Message quote swipe action

**Testing**
- More db upgrades. We expect some more crashes. Please provide steps how to reproduce if a crash occurs.
- OMEMO error handling after appex / main app race conditions

**TODO**:
- Connect Auto Download sliders to auto download code
- Groups

### CODE REVIEW

File: Monal/Classes/DataLayer.m [hunk 1]
```diff
@@ -483,10 +487,10 @@ -(MLContact*) contactForUsername:(NSString*) username forAccount:(NSString*) acc
 
 -(NSMutableArray<MLContact*>*) contactList
 {
-    //only list contacts having a roster entry (e.g. kSubBoth, kSubTo or kSubFrom)
-    NSString* query = @"SELECT buddy_name, a.account_id, IFNULL(IFNULL(NULLIF(A.nick_name, ''), NULLIF(A.full_name, '')), buddy_name) AS 'sortkey' FROM buddylist AS A INNER JOIN account AS b ON a.account_id=b.account_id WHERE (a.subscription=? OR a.subscription=? OR a.subscription=?) AND b.enabled=1 AND (b.username || '@' || b.domain) != buddy_name ORDER BY sortkey COLLATE NOCASE ASC;";
+    //list all contacts without groups
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/DataLayer.m:490]: what does "ithout grups" mean? roster groups? why without roster grops?

File: Monal/Classes/chatViewController.m [hunk 2]
```diff
@@ -1879,7 +1884,7 @@ -(UITableViewCell*) tableView:(UITableView*) tableView cellForRowAtIndexPath:(NS
         }
         else // Default case
         {
-            if(messageText.length ==2  && [EmojiUtilities containsEmoji:messageText]){
+            if(messageText.length == 2  && [MLEmoji containsEmojiWithText:messageText]) {
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/chatViewController.m:1887]: why length == 2 does this count bytes or utf8 symbols? for utf8 symbols it should be 1 or 2 or 3 (ex modifier and skin tone modifier)...for bytes it should be an entirely different test...

File: Monal/Classes/chatViewController.m [hunk 3]
```diff
@@ -2114,6 +2123,27 @@ -(UISwipeActionsConfiguration*) tableView:(UITableView*) tableView trailingSwipe
     {
         LMCEditAction.image = [[[UIImage systemImageNamed:@"pencil.circle.fill"] imageWithHorizontallyFlippedOrientation] imageWithTintColor:UIColor.whiteColor renderingMode:UIImageRenderingModeAutomatic];
     }
+
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/chatViewController.m:2126]: could this be added as swipe right action instead as swipe left which is already crowded with other actions?

**Review by tmolitor-stud-tu — CHANGES_REQUESTED** [review]

File: Monal/Classes/chatViewController.m (hunk 3 above)
Comment (FriedrichAltheide) [reply Monal/Classes/chatViewController.m:2126]: @janst97 Could you have a look at this for the next beta release?

File: Monal/Classes/chatViewController.m (hunk 2 above)
Comment (FriedrichAltheide) [reply Monal/Classes/chatViewController.m:1887]: @anurodhp Could you have a look at this for the next beta release?

File: Monal/Classes/DataLayer.m (hunk 1 above)
Comment (FriedrichAltheide) [reply Monal/Classes/DataLayer.m:490]: @FriedrichAltheide TODO for next beta

**Review by tmolitor-stud-tu — APPROVED** [review]

**Pull request merged**